app/models/audit_activity/risk_assessment/risk_assessment_updated.rb

```diff
@@ -0,0 +1,126 @@
+class AuditActivity::RiskAssessment::RiskAssessmentUpdated < AuditActivity::Base
+  def self.from(*)
+    raise "Deprecated - use AddRiskAssessmentToCase.call instead"
+  end
+
+  def self.build_metadata(risk_assessment:, previous_product_ids:, previous_attachment_filename:)
+    updates = risk_assessment.previous_changes.slice(
+      :assessed_on,
+      :risk_level,
+      :custom_risk_level,
+      :assessed_by_team_id,
+      :assessed_by_business_id,
+      :assessed_by_other,
+      :details
+    )
+
+    if previous_product_ids.sort != risk_assessment.product_ids.sort
+      updates[:product_ids] = [previous_product_ids, risk_assessment.product_ids]
+    end
+
+    current_attachment_filename = risk_assessment.risk_assessment_file.filename
+
+    if previous_attachment_filename != current_attachment_filename
+      updates["filename"] = [previous_attachment_filename, current_attachment_filename]
+    end
+
+    {
+      risk_assessment_id: risk_assessment.id,
+      updates: updates
+    }
+  end
+
+  def risk_level_changed?
+    new_risk_level || new_custom_risk_level
+  end
+
+  def assessed_by_changed?
+    new_assessed_by_team_id || new_assessed_by_business_id || new_assessed_by_other
+  end
+
+  def products_changed?
+    new_product_ids
+  end
+
+  def new_assessed_on
+    updates["assessed_on"]&.second
+  end
+
+  def new_risk_level
+    updates["risk_level"]&.second
+  end
+
+  def new_filename
+    updates["filename"]&.second
+  end
+
+  def new_assessed_by_team
+    if new_assessed_by_team_id
+      Team.find(new_assessed_by_team_id)
+    end
+  end
+
+  def new_assessed_by_business
+    if new_assessed_by_business_id
+      Business.find(new_assessed_by_business_id)
+    end
+  end
+
+  def new_assessed_by_team_id
+    updates["assessed_by_team_id"]&.second
```

Comment:
i've done this for the corrective action 
```suggestion
    updates.dig("assessed_by_team_id", 1)
```